### ISSUE TITLE
Call to a member function metadata() on null

### ISSUE THREAD

**Issue by wdebusschere** [tobimori/kirby-seo#66]
I added seo/page to the page blueprint (also added the seo/site to the site blueprint), and it gives the error Call to a member function metadata() on null. the element with the error:

```
<div class="k-field k-info-field" form-data="[object Object]" name="robots" signature="f4ec69e10a563a018e3f8047a2b187f7" strict="true" type="info" width="1/1" section="seo-col-main-fields" endpoints="[object Object]"><h2 class="k-headline">Error</h2><div data-align="start" data-theme="negative" class="k-box"><!----><div class="k-text"><p>Call to a member function metadata() on null</p></div></div><!----></div>
```

![Screenshot 2024-01-30 at 09 32 52](https://github.com/tobimori/kirby-seo/assets/4136426/b8f23c8e-c693-4aa9-8cbd-c0607d91803e)

**Comment by tobimori** [comment]
Are you on 1.0.0-rc.1?

**Comment by wdebusschere** [comment]
"tobimori/kirby-seo": "1.0.0-rc.1"

**Comment by tobimori** [comment]
Interesting, I had the same error and fixed it in rc.1 - but it seems I didn't catch all edge cases.

Why this error happens:
The robots blueprint is a dynamic blueprint that reads the request path (e.g. panel/pages/page+subpage+etc/some-subroute) and tries to find the page based on the middle part so it can correctly show the placeholder/fallback. 
This error happens when it is unable to find a page.

In rc.1, I fixed that draft pages could also be found. 

Is there anything special about this page or your environment that might lead to this issue? Can you share the request URLs from the network tab?

**Comment by wdebusschere** [comment]
$kirby->request()->path()->data()[2]
on my localhost environment i just access the panel: 
http://localhost/domain.pt/public/panel/pages/clients?tab=seo

**Comment by tobimori** [comment]
i assume it's related to running on a subfolder then, will check

**Comment by whitmoa** [comment]
Hi, Tobias. Any update on this issue? I'm getting the same problem with 1.0.0-rc.2 in a localhost installation and a public-facing UAT environment. Both of them run in subfolders.

Thanks.

**Comment by tobimori** [comment]
I could not reproduce this so far with my setup, have to try something else

**Comment by whitmoa** [comment]
For what it's worth, I've still got a previous version of the plugin that was just called 1.0.0 – no 'rc' release reference. With that version, the Robots blueprint section worked OK within seo/page. But it wasn't generating a sitemap.

With 1.0.0-rc.2, the sitemap generation works but, because of the issue with the Robots section in the blueprint, it is empty.

**Comment by wdebusschere** [comment]
@tobimori That is the code i'm using, i does the same like your code.

**Comment by tobimori** [comment]
Hi, this should be fixed here: https://github.com/tobimori/kirby-seo/commit/36d05382e29536b12af140280ec640d51f09c008

Would appreciate if this actually fixes your issue by testing the mainbranch before I release it! Thanks :)

**Comment by wdebusschere** [comment]
will test now

**Comment by tobimori** [comment]
Just released it as 1.0.0-rc.3 then.

**Comment by quentin-f451** [comment]
This is actually not working when page `uid` has a dash in it.
https://github.com/tobimori/kirby-seo/blob/c9e42c595c540a87791f32c9cf5d478d6b5a3cab/blueprints/fields/robots.php#L24
I would modify this line to:
```php
$matches = Str::match($path, "/pages\/([a-zA-Z0-9-_+]+)\/?/m");
```

**Comment by tobimori** [comment]
you're right, completely forgot that! thanks.

**Comment by tobimori** [comment]
Is the page you're trying to access a draft? It seems like there's a Kirby bug in that case.

**Comment by LuckyRessource** [comment]
No, it is not a draft page. It is a product page that already existed before the plugin was installed.

**Comment by tobimori** [comment]
Is any page in the chain a draft? I can only reproduce it that way

**Comment by tobimori** [comment]
okay, let me know if you have anything else that might in reproducing this error.

**Comment by tobimori** [comment]
As said, this error is from Kirby core and I'll wait for a fix from them

**Comment by quentin-f451** [comment]
Might be a good idea to remove the required property from the Robots directives part until the correction arrives from Kirby side? This makes the plugin quite annoying to use as drafts don't get the default value from parent. Is there any reason for these fields to be required?

**Comment by tobimori** [comment]
1.0.0 should fix the error.
